BUG: When ITK_BUILD_MINC_SUPPORT enabled, antsRegistration can't write out non-minc warp transform files

Calling @vfonov who implemented this

Related to #264 I believe.

If I build plain ANTs superbuild without ITK_BUILD_MINC_SUPPORT=OFF, antsRegistration works as expected.

If I set ITK_BUILD_MINC_SUPPORT=ON, antsRegistration will write out MINC-style transforms when ``--minc`` is specified on the command line.

Similarly, without ``--minc`` antsRegistration will successfully write out ``.mat`` files for affine-like transforms.

However, when antsRegistration tries to write the warp transform:
```
 Can't write transform file test1Warp.nii.gz
 Exception Object caught: 
 
 itk::ImageFileWriterException (0x4d54690)
 Location: "unknown" 
 File: /opt/quarantine/ANTs/git/src/build/ITKv4-install/include/ITK-4.9/itkImageFileWriter.hxx
 Line: 151
 Description:  Could not create IO object for writing file test1Warp.nii.gz
   Tried to create one of the following:
     NiftiImageIO
     NrrdImageIO
     GiplImageIO
     HDF5ImageIO
     JPEGImageIO
     GDCMImageIO
     BMPImageIO
     LSMImageIO
     PNGImageIO
     TIFFImageIO
     VTKImageIO
     StimulateImageIO
     BioRadImageIO
     MetaImageIO
     MRCImageIO
     GE4ImageIO
     GE5ImageIO
     MINCImageIO
     MGHImageIO
   You probably failed to set a file suffix, or
     set the suffix to an unsupported type. Can't write transform file test1Warp.nii.gz
 Exception Object caught: 
```

I'm looking through PR #202 trying to find where the confusion might be generated, but I can't see it.

Weirdly, when I update the git commit id in the ANTs superbuild to that and build, I run into this error, even though it passed the build bot on ITK:
```
[ 90%] Building C object Modules/ThirdParty/MINC/src/libminc/CMakeFiles/itkminc2.dir/libsrc2/hyper.c.o
/home/cic/devgab/projects/ANTs/build/ITKv4/Modules/ThirdParty/MINC/src/libminc/libsrc2/hyper.c:64:1: error: redefinition of ‘round’
 round(double x)
 ^
make[5]: *** [Modules/ThirdParty/MINC/src/libminc/CMakeFiles/itkminc2.dir/libsrc2/hyper.c.o] Error 1
make[4]: *** [Modules/ThirdParty/MINC/src/libminc/CMakeFiles/itkminc2.dir/all] Error 2
```

Any thoughts?

basically it means that cmake could not detect presense of round function,
for some reason.


On Fri, Jan 15, 2016 at 4:43 PM, Gabriel A. Devenyi <
notifications@github.com> wrote:

> Weirdly, when I update the git commit id in the ANTs superbuild to that
> and build, I run into this error, even though it passed the build bot on
> ITK:
>
> [ 90%] Building C object Modules/ThirdParty/MINC/src/libminc/CMakeFiles/itkminc2.dir/libsrc2/hyper.c.o
> /home/cic/devgab/projects/ANTs/build/ITKv4/Modules/ThirdParty/MINC/src/libminc/libsrc2/hyper.c:64:1: error: redefinition of ‘round’
>  round(double x)
>  ^
> make[5]: *** [Modules/ThirdParty/MINC/src/libminc/CMakeFiles/itkminc2.dir/libsrc2/hyper.c.o] Error 1
> make[4]: *** [Modules/ThirdParty/MINC/src/libminc/CMakeFiles/itkminc2.dir/all] Error 2
>
> Any thoughts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stnava/ANTs/issues/270#issuecomment-172102668>.
>



-- 
Best regards,

 Vladimir S. Fonov ~ vladimir <dot> fonov <at> gmail <dot> com

--- a/SuperBuild/External_ITKv4.cmake
+++ b/SuperBuild/External_ITKv4.cmake
@@ -161,8 +161,8 @@ if(NOT DEFINED ${extProjName}_DIR AND NOT ${USE_SYSTEM_${extProjName}})
 
   ### --- End Project specific additions
   set(${proj}_REPOSITORY ${git_protocol}://github.com/InsightSoftwareConsortium/ITK.git)
-  set(${proj}_GIT_TAG fc03bcc92b11326b87c232c14eb234fc1f791f9f) ##
-  set(ITK_VERSION_ID ITK-4.9) ### NOTE: When updating GIT_TAG, also update ITK_VERSION_ID
+  set(${proj}_GIT_TAG f60bbd3aadc4b9fbf2cff1cf7227cef1e92b979a) ##
+  set(ITK_VERSION_ID ITK-4.10) ### NOTE: When updating GIT_TAG, also update ITK_VERSION_ID
 
   ExternalProject_Add(${proj}
     GIT_REPOSITORY ${${proj}_REPOSITORY}

